webapp/channels/src/components/user_settings/notifications/send_test_notification_notice.tsx

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useMemo, useRef, useState} from 'react';
+import {useIntl} from 'react-intl';
+
+import {sendTestNotification} from 'actions/notification_actions';
+
+import SectionNotice from 'components/section_notice';
+
+const sectionNoticeContainerStyle: React.CSSProperties = {marginTop: 20};
+
+const onGoToNotificationDocumentation = () => {
+    window.open('https://docs.mattermost.com/collaborate/mention-people.html');
+};
+
+const TIME_TO_SENDING = 500;
+const TIME_TO_SEND = 500;
+const TIME_TO_IDLE = 3000;
+
+type Props = {
+    adminMode?: boolean;
+};
+
+const SendTestNotificationNotice = ({
+    adminMode = false,
+}: Props) => {
+    const intl = useIntl();
+    const [buttonState, setButtonState] = useState<'idle'|'sending'|'sent'|'error'>('idle');
```

Comment:
Nit: Break out the type for readability. Or better yet, if it's just scoped to this file, use an `enum`

Reply:
Typescript works better with these kind of types than with enums, as far as I know, so let's keep this kind of type. Nevertheless, I extracted it to the outside.

That being said, I would like to do a "SmartSpinningButton" that does all this logic for us, but that is when I have my button component PR merged :P